src/vm/types/serialization.rs

```diff
@@ -702,28 +702,6 @@ impl ClarityDeserializable<Value> for Value {
     }
 }
 
-impl ClaritySerializable for u32 {
-    fn serialize(&self) -> String {
-        let mut buffer = Vec::new();
-        buffer
-            .write_all(&self.to_be_bytes())
-            .expect("u32 serialization: failed writing.");
-        to_hex(buffer.as_slice())
-    }
-}
-
-impl ClarityDeserializable<u32> for u32 {
-    fn deserialize(input: &str) -> Self {
-        let bytes = hex_bytes(&input).expect("u32 deserialization: failed decoding bytes.");
-        assert_eq!(bytes.len(), 4);
-        u32::from_be_bytes(
-            bytes[0..4]
-                .try_into()
-                .expect("u32 deserialization: failed reading."),
-        )
-    }
-}
```

Comment:
This needs to be preserved, and the `clarity_serializable!(u32);` macro invocation needs to be removed: these serialization methods are already used in Stacks 2.05, so switching how u32's are serialized in Clarity (which this change would do) would be consensus-breaking to 2.05.

Reply:
Yikes, I can't believe I missed that.  Yeah, in general, *any* code that does serialization or deserialization needs to be treated as consensus-critical.